src/app/tabs/camera.cpp

```diff
@@ -0,0 +1,83 @@
+#include <app/config.hpp>
+#include <app/tabs/camera.hpp>
+#include <app/window.hpp>
+#include <app/theme.hpp>
+
+CameraTab::CameraTab(QWidget *parent) : QWidget(parent)
+{
+    settings = Config::get_instance()->get_settings();
+
+    QVBoxLayout *layout = new QVBoxLayout(this);
+
+	player.setVideoOutput(&videoWidget);
+	QString camera = settings->value("cameraName").toString();
+	cameraName = new QLabel(camera, this);
+
+    cameraName->setFont(Theme::font_16);
+   	status.setFont(Theme::font_14);
+
+	layout->addWidget(cameraName);
+	layout->addWidget(&videoWidget);
+	layout->addWidget(&status);
+
+	connect(&player, &QMediaPlayer::mediaStatusChanged, this, &CameraTab::ChangedStatus);
+	connect_stream();
+}
+
+void CameraTab::ChangedStatus(QMediaPlayer::MediaStatus mediaStatus)
+{
+    qInfo() << "Camera status changed to: " << mediaStatus;

```

Comment:
I left this in deliberately because I don't fully understand the scenarios under which the various statuses happen.